cpp/src_prims/metrics/adjustedRandIndex.h

```diff
@@ -91,64 +92,78 @@ double computeAdjustedRandIndex(
 
   //creating device buffers for all the parameters involved in ARI calculation
   //device variables
-  MLCommon::device_buffer<int> a(allocator, stream, numUniqueClasses);
-  MLCommon::device_buffer<int> b(allocator, stream, numUniqueClasses);
-  MLCommon::device_buffer<int> d_aCTwoSum(allocator, stream, 1);
-  MLCommon::device_buffer<int> d_bCTwoSum(allocator, stream, 1);
-  MLCommon::device_buffer<int> d_nChooseTwoSum(allocator, stream, 1);
+  MLCommon::device_buffer<unsigned long long int> a(allocator, stream,
+                                                    numUniqueClasses);
+  MLCommon::device_buffer<unsigned long long int> b(allocator, stream,
+                                                    numUniqueClasses);
+  MLCommon::device_buffer<unsigned long long int> d_aCTwoSum(allocator, stream,
+                                                             1);
+  MLCommon::device_buffer<unsigned long long int> d_bCTwoSum(allocator, stream,
+                                                             1);
+  MLCommon::device_buffer<unsigned long long int> d_nChooseTwoSum(allocator,
+                                                                  stream, 1);
   //host variables
-  int h_aCTwoSum;
-  int h_bCTwoSum;
-  int h_nChooseTwoSum;
+  double h_aCTwoSum;

```

Comment:
This cannot be `double`, because this variable needs to retrieve the reduction value from device. Later on, during the final computation of adjustedRandIndex, type-casting it to `double` (or `long double`) is the right way to go.